pkg/multus/multus.go

```diff
@@ -583,50 +606,148 @@ func CmdAdd(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) (c
 		return nil, cmdErr(nil, "error getting k8s args: %v", err)
 	}
 
-	if n.ReadinessIndicatorFile != "" {
+	pod, err := getPod(kubeClient, k8sArgs, false)
+	if err != nil {
+		return nil, err
+	}
+
+	netnsPath := args.Netns
+	if netns, found := os.LookupEnv("CNI_NETNS"); found {
+		netnsPath = netns
+	}
+
+	ifName := args.IfName
+	if interfaceName, found := os.LookupEnv("CNI_IFNAME"); found {
+		ifName = interfaceName
+	}
+	result, err := NewMultusCmd(
+		args.ContainerID,
+		string(k8sArgs.K8S_POD_INFRA_CONTAINER_ID),
+		ifName,
+		netnsPath,
+		string(k8sArgs.K8S_POD_NAME),
+		string(k8sArgs.K8S_POD_NAMESPACE),
+		string(k8sArgs.K8S_POD_UID)).Add(n, pod, exec, kubeClient)
+	if err != nil {
+		return nil, fmt.Errorf("%v", err)
+	}
+	return result, nil
+}
+
+//CmdCheck ...
+func CmdCheck(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) error {
+	in, err := types.LoadNetConf(args.StdinData)
+	logging.Debugf("CmdCheck: %v, %v, %v", args, exec, kubeClient)
+	if err != nil {
+		return err
+	}
+
+	k8sArgs, err := k8s.GetK8sArgs(args)
+	if err != nil {
+		return cmdErr(nil, "error getting k8s args: %v", err)
+	}
+
+	return NewMultusCmd(
+		args.ContainerID,
+		string(k8sArgs.K8S_POD_INFRA_CONTAINER_ID),
+		args.IfName,
+		args.Netns,
+		string(k8sArgs.K8S_POD_NAME),
+		string(k8sArgs.K8S_POD_NAMESPACE),
+		string(k8sArgs.K8S_POD_UID)).Check(in, exec)
+}
+
+//CmdDel ...
+func CmdDel(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) error {
+	in, err := types.LoadNetConf(args.StdinData)
+	logging.Debugf("CmdDel: %v, %v, %v", args, exec, kubeClient)
+	if err != nil {
+		return err
+	}
+
+	k8sArgs, err := k8s.GetK8sArgs(args)
+	if err != nil {
+		return cmdErr(nil, "error getting k8s args: %v", err)
+	}
+
+	kubeClient, err = k8s.GetK8sClient(in.Kubeconfig, kubeClient)
+	if err != nil {
+		return cmdErr(nil, "error getting k8s client: %v", err)
+	}
+
+	pod, err := getPod(kubeClient, k8sArgs, true)
+	if err != nil {
+		return err
+	}
+
+	return NewMultusCmd(
+		args.ContainerID,
+		string(k8sArgs.K8S_POD_INFRA_CONTAINER_ID),
+		args.IfName,
+		args.Netns,
+		string(k8sArgs.K8S_POD_NAME),
+		string(k8sArgs.K8S_POD_NAMESPACE),
+		string(k8sArgs.K8S_POD_UID),
+	).Delete(in, pod, exec, kubeClient)
+}
+
+// NewMultusCmd return a new ADD / CHECK / DEL command
+func NewMultusCmd(containerID, sandboxID, ifaceName, netnsPath, podName, podNamespace, podUID string) *Cmd {
+	return &Cmd{
+		ContainerID:   containerID,
+		SandboxID:     sandboxID,
+		InterfaceName: ifaceName,
+		NetnsPath:     netnsPath,
+		PodName:       podName,
+		PodNamespace:  podNamespace,
+		PodUID:        podUID,
+	}
+}
+
+// Add is the handler of the CNI ADD command
+func (mc *Cmd) Add(cniConf *types.NetConf, pod *v1.Pod, exec invoke.Exec, kubeclient *k8s.ClientInfo) (cnitypes.Result, error) {
```

Comment:
Could you please change it to `Add(mc *Cmd, cniConf *types.NetConf, pod *v1.Pod, exec invoke.Exec, kubeclient *k8s.ClientInfo) (cnitypes.Result, error) {`?

Reply:
Done.